### PR TITLE
Feature/improvenance

### PR DESCRIPTION
**Pull request by Tooni**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [ ] Make sure you are requesting to **pull a feature/bugfix branch** (right side). Don't request from master!
- [ ] Check the commit's or even all commits' message styles matches our requested structure.
- [ ] Check your code additions will fail neither code linting checks nor unit test.

### Description
Added new provenance events and displayed them in the UI. Also ordered all provenance blocks by their date string fields. The events themselves are not accessible by the UI, you need to use truffle console.

image

### Checklist
* [x] Have you done your changes on a separate branch. Branches MUST have descriptive names that start with either the `fix/` or `feature/` prefixes. Good examples are: `fix/signin-loop` or `feature/pr-templates`.
* [x] Have you got descriptive commit messages that would make sense if prefixed with "This commit will ..."
* [x] Will you squash when you merge this PR?

#### New Features

1. [ ] Does your submission have a high coverage of your feature?
2. [x] Have you lint your code locally prior to submission?

💔Thank you!

### CODE REVIEW

File: app/src/components/Provenance.tsx [hunk 1]
```diff
@@ -117,7 +148,12 @@ const ProvenanceTimeline: React.FC<{records: ProvenanceRecord[]}> = ({ records }
               <AddressField label='Artist' address={record.artist}/>
             </Form>
             : null}
-
+          {record.type !== 'mint' && record.type !== 'sale' && record.detail
+            ? <Form>
+              <PlaintextField label='Date' value={record.detail.date} />
+              <PlaintextField label='Detail' value={record.detail.detailInfo} />
+            </Form>
+            : null}
```
Comment (JamesDalboth) [inline app/src/components/Provenance.tsx]: instead of `cond ? x : null` we can do `cond && x`

File: app/src/components/Provenance.tsx [hunk 2]
```diff
@@ -148,27 +184,84 @@ export const Provenance: React.FC<{tokenId: number}> = ({ tokenId }) => {
...
+        }
+        if (!aDate && bDate) {
+          return 1;
+        }
+        if (aDate && bDate) {
+          return aDate.localeCompare(bDate);
+        }
+      }
+      return -1;
+    };
```
Comment (JamesDalboth) [inline app/src/components/Provenance.tsx:258]: If we are sorting the records surely the first thing we should be inspecting is date not artists?

**Review by JamesDalboth — CHANGES_REQUESTED** [review]
1 Small change and one thing I am unsure about, I may be wrong but if right needs to be fixed

File: app/src/components/Provenance.tsx (hunk 2 above)
Comment (Tooni) [reply app/src/components/Provenance.tsx:258]: the way the provenance record is used means that records that have artists (aka registration records) are guaranteed to not have dates, and vice versa, so it's unclear where to put such records in the timeline. ive arbitrarily put them at the top

Comment (JamesDalboth) [reply app/src/components/Provenance.tsx:258]: okay makes sense!

Comment (Tooni) [reply app/src/components/Provenance.tsx:258]: the only record this applies to is the one for when a piece is first registered

**Review by JamesDalboth — APPROVED** [review]

**Pull request merged**